_posts/2018-12-11-dipto-csc-799-report.md

```diff
@@ -0,0 +1,234 @@
+
+Let's see a very popular example.
+
+If we know a pair of words that are from same group like (man, king) and we want to get a similar pair where we know one entry like this: (woman, ?).
+
+A standard word embedding model would suggest a like queen as the answer. Because it is the closest word that can replace king when woman replace the word man.
+
+Something like this: queen = (king - man) + woman
+
+Popular word embedding models include word2vec model by Google, GloVe by Stanford. There are some unconventional word embedding models available as well like urban dictionary model.
+
+Download links for the pretrained
+* GoogleNews word2vec model:  https://drive.google.com/file/d/0B7XkCwpI5KDYNlNUTTlSS21pQmM/edit
+* GloVe model: http://nlp.stanford.edu/data/glove.6B.zip
+* Urban Dictionary word2vec model: https://data.world/jaredfern/urban-dictionary-embedding
+
+The model that you need to use depends on the application that you have in mind.
+
+We can easily find already trained models for word2vec or Glove easily. However, you may need to develop/train the model from corpus at hand if you have a very specific application field in mind. For example, Google's word2vec model was trained with entire Wikipedia corpus. The basic idea is similar to n-gram approach that means we need a group of words at a time to get idea of the context of the use of the word. Here, I show a general step through approach for training your word embedding model in Gensim, a more matured natural language processing framework than natural language toolkit (NLTK) that comes readily with python.
+
+```python
+sentences = 'This is a small corpora'	#assume we fit all our corpus in a variable in memory
+model = Word2Vec(sentences)
+
+words = list(model.wv.vocab)	#vocabulary of the corpora
+model.wv.save_word2vec_format('my_word_embedding_model', binary = True)
+```
+
+We can load a pre-trained model by using:
+
+```python
+model = Word2Vec.load('my_word_embedding_model')
+```
+
+If we have a group of words meaning similar thing and we know only one or two meaning the opposite, word embedding models can find the similar words; that can even specialized for our problem context if we chose to train our own word embedding models.
+
+Upto this point we have talking about how we can use words or synonyms of words to perform natural language processing tasks. These sort of approaches work better with naive Bayes distribution model when the vocabulary for problem space is known and predefined. In case where we do not want our system to be limited by the EXACT words in the corpus rather we want to have an overall idea about the way of expression or conveying of information, we use sentiment analysis. Sentiment analysis being a popular technique comes readily with python NLTK library. Besides, a good number of resouces can be found on the internet about it. In this report, we will discuss a relatively new approach called tone analysis. 
```

Comment:
```suggestion
Upto this point I have talked about how we can use words or synonyms of words to perform natural language processing tasks. These sort of approaches work better with naive Bayes distribution model when the vocabulary for problem space is known and predefined. In case where we do not want our system to be limited by the EXACT words in the corpus rather we want to have an overall idea about the way of expression or conveying of information, we use sentiment analysis. Sentiment analysis being a popular technique comes readily with python NLTK library. Besides, a good number of resources can be found on the internet about it. In this report, we will discuss a relatively new approach called tone analysis. 
```